doit/dependency.py

```diff
@@ -426,6 +426,26 @@ def get_state(self, dep, current_state):
             'timestamp': TimestampChecker}
 
 
+class DependencyStatus(object):
+    """Result object for Dependency.get_status.
+
+    @ivar status: (str) one of "run", "up-to-date" or "error"
+    """
+
+    def __init__(self):
+        self.status = None
+
+    def _decide(self, status):
+        """Sets state of status object if it wasn't set before."""
+        if self.status is None:
+            self.status = status
+
+    def raise_exception(self):
+        """Raises a DependencyException if the state is 'error'."""
+        if self.status == "error":
+            raise DependencyException("Dependent file '{}' does not exist.".format(self._error_reason))
```

Comment:
I mean this exception could be completely removed. Where this exception is used, it could just deal with the return value.